### ISSUE TITLE
Buffered context means no streaming support

### ISSUE THREAD

**Issue by Marak** [fission/fission#93]
`fission` appears to use a buffered context for both receiving and sending data.

This prevents the ability to create streaming transformations, or process any data coming into the function as a stream. Do you have any plans to add streaming support? Seems like a pretty big thing to not have

`microcule` has streaming support ( as well as a plugin system and support for 14+ languages ) https://github.com/stackvana/microcule

**Comment by ktrance** [comment]
:+1:

**Comment by soamvasani** [comment]
Yes, we're interested in the streaming use case.  Changing the title to be a bit more general.  I *think* this will only need changes in the environments, since the router is just a proxy; but we'll have to test it out.

Also, I see you've linked microcule a couple times in the issues -- do you work on that project?  Is your goal here some sort of collaboration?  Happy to discuss more.

**Comment by Marak** [comment]
`microcule` is a project I built over the past two years to support https://hook.io

I have strong experience and knowledge in this domain.

My main interest is making sure that this project ( `fission` ) is adequately covering what I feel is the Minimal Viable Product for an open-source functions as a service platform.

**Comment by schmurfy** [comment]
has anything change ?
We are considering a usecase where we would need to transfer a large volume of data between two functions, http should work well for this if supported.

**Comment by sanketsudake** [comment]
This has been inactive for 1653 days and is being marked stale. It will be closed in a future scrub if there's no further activity. Comment to keep it open.

**Comment by sanketsudake** [comment]
Streaming function invocation is supported natively now — set `streaming` on a Function's spec (with `streamingProtocol` to choose the upstream mode) and the router proxies the response as a stream.

Closing as completed; please reopen with details if your use case isn't covered on a current release.
